server/src/test/java/org/elasticsearch/snapshots/CorruptedBlobStoreRepositoryIT.java

```diff
@@ -0,0 +1,201 @@
+
+public class CorruptedBlobStoreRepositoryIT extends AbstractSnapshotIntegTestCase {
+
+    public void testConcurrentlyChangeRepositoryContents() throws Exception {
+        Client client = client();
+
+        Path repo = randomRepoPath();
+        final String repoName = "test";
+
+        logger.info("-->  creating repository at {}", repo.toAbsolutePath());
+        execute("CREATE REPOSITORY test TYPE fs with (location=?, compress=false, chunk_size=?)",
+                new Object[]{repo.toAbsolutePath().toString(), randomIntBetween(100, 1000) + ByteSizeUnit.BYTES.getSuffix()});
+
+        execute("create table doc.test1(x integer)");
+        execute("create table doc.test2(x integer)");
+
+        logger.info("--> indexing some data");
+
+        execute("insert into doc.test1 values(1),(2)");
+        execute("insert into doc.test2 values(1),(2)");
+
+        final String snapshot = "snapshot1";
+
+        logger.info("--> creating snapshot");
+        CreateSnapshotResponse createSnapshotResponse = client.admin().cluster().prepareCreateSnapshot(repoName, snapshot)
+            .setWaitForCompletion(true).setIndices("test*").get();
+        assertThat(createSnapshotResponse.getSnapshotInfo().successfulShards(), greaterThan(0));
+        assertThat(createSnapshotResponse.getSnapshotInfo().successfulShards(),
+                   equalTo(createSnapshotResponse.getSnapshotInfo().totalShards()));
+
+        logger.info("--> move index-N blob to next generation");
+        final RepositoryData repositoryData =
+            getRepositoryData(internalCluster().getMasterNodeInstance(RepositoriesService.class).repository(repoName));
+        Files.move(repo.resolve("index-" + repositoryData.getGenId()), repo.resolve("index-" + (repositoryData.getGenId() + 1)));
+
+        assertRepositoryBlocked(client, repoName, snapshot);
+
+        if (randomBoolean()) {
+            logger.info("--> move index-N blob back to initial generation");
+            Files.move(repo.resolve("index-" + (repositoryData.getGenId() + 1)), repo.resolve("index-" + repositoryData.getGenId()));
+
+            logger.info("--> verify repository remains blocked");
+            assertRepositoryBlocked(client, repoName, snapshot);
+        }
+
+        logger.info("--> remove repository");
+        execute("drop REPOSITORY test");
+
+        logger.info("--> recreate repository");
+
+        execute("CREATE REPOSITORY test TYPE fs with (location=?, compress=false, chunk_size=?)",
+                new Object[]{repo.toAbsolutePath().toString(), randomIntBetween(100, 1000) + ByteSizeUnit.BYTES.getSuffix()});
+
+        logger.info("--> delete snapshot");
+        execute("drop snapshot test.snapshot1");
+
+        logger.info("--> make sure snapshot doesn't exist");
+        expectThrows(SnapshotMissingException.class, () -> client.admin().cluster().prepareGetSnapshots(repoName)
+            .addSnapshots(snapshot).get().getSnapshots());
+    }
+
+    @Repeat(iterations = 20)
```

Comment:
Is this a leftover or intentional?

Reply:
Ups, my bad forgot to push the last changes.